integration_tests/test_solo_machine.py

```diff
@@ -0,0 +1,179 @@
+    def __init__(self, temp_path, base_port=26650, chain_id="devnet-solomachine"):
+        self.chain_id = chain_id
+        self.grpc_port = ports.grpc_port(base_port)
+        self.rpc_port = ports.rpc_port(base_port)
+        self.solomachine_home = os.path.join(os.environ["SOLO_MACHINE_HOME"])
+        print(self.solomachine_home)
+        self.bin_file = os.path.join(self.solomachine_home, "solo-machine")
+
+        os_platform = platform.system()
+        if os_platform == "Darwin":
+            self.sign_file = os.path.join(
+                self.solomachine_home, "libmnemonic_signer.dylib"
+            )
+        else:
+            self.sign_file = os.path.join(
+                self.solomachine_home, "libmnemonic_signer.so"
+            )
+        self.trusted_height = None
+        self.trusted_hash = None
+        self.db_path = f'sqlite://{temp_path.join("solo-machine.db")}'
+
+    def get_chain_info(self):
+        result = requests.get(f"http://127.0.0.1:{self.rpc_port}/block").json()
+        self.trusted_hash = result["result"]["block_id"]["hash"]
+        self.trusted_height = result["result"]["block"]["header"]["height"]
+
+    def set_env(self):
+        os.environ["SOLO_DB_URI"] = self.db_path
+        os.environ["SOLO_SIGNER"] = self.sign_file
+        os.environ[
+            "SOLO_MNEMONIC"
+        ] = "awesome there minute cash write immune tag \
+reopen price congress trouble reunion south wisdom donate credit below leave \
+wisdom eagle sail siege rice train"
+        os.environ["SOLO_HD_PATH"] = "m/44'/394'/0'/0/0"
+        os.environ["SOLO_ACCOUNT_PREFIX"] = "cro"
+        os.environ["SOLO_ADDRESS_ALGO"] = "secp256k1"
+        os.environ["SOLO_FEE_AMOUNT"] = "1000"
+        os.environ["SOLO_FEE_DENOM"] = "basecro"
+        os.environ["SOLO_GAS_LIMIT"] = "300000"
+        os.environ["SOLO_GRPC_ADDRESS"] = f"http://0.0.0.0:{self.grpc_port}"
+        os.environ["SOLO_RPC_ADDRESS"] = f"http://0.0.0.0:{self.rpc_port}"
+        os.environ["SOLO_TRUSTED_HASH"] = self.trusted_hash
+        os.environ["SOLO_TRUSTED_HEIGHT"] = self.trusted_height
+
+    def prepare(self):
+        self.get_chain_info()
+        self.set_env()
+
+    def run_sub_cmd(self, sub_cmd, json_output=True):
+        if json_output:
+            cmd = f"{self.bin_file} --output json {sub_cmd}"
+        else:
+            cmd = f"{self.bin_file} {sub_cmd}"
+        output = interact(cmd)
+        data = parse_output(output, json_output)
+        return data
+
+    def init(self):
+        sub_cmd = "init"
+        return self.run_sub_cmd(sub_cmd)
+
+    def add_chain(self):
+        sub_cmd = "chain add"
+        return self.run_sub_cmd(sub_cmd)
+
+    def connect_chain(self):
+        sub_cmd = f"ibc connect {self.chain_id}"
+        result = self.run_sub_cmd(sub_cmd, False)
+        if "error" in result:
+            raise Exception(result)
+
+    def get_balance(self):
+        sub_cmd = f"chain balance {self.chain_id} solotoken"
```

Comment:
I guess "solotoken" is the denomination? Can it be extracted into a constant or an argument/config parameter?